site/compare.md

```diff
@@ -0,0 +1,284 @@
+                        DefaultTargetingEnabled: true
+                        JNDIName: jms/myUDQ
+                        ResetDeliveryCountOnForward: true
+```
+To compare the two model files, run the tool as follows:
+
+    weblogic-deploy\bin\compareModel.cmd -oracle_home c:\wls12213 new_model.yaml old_model.yaml
+
+The output of the tool will look something like this:
+
+```
+Comparing Models: new=/tmp/model2.yaml vs old=/tmp/model1.yaml
+
+Differences between new model and old model:
+
+resources:
+    JMSSystemResource:
+        MyJmsModule:
+            JmsResource:
+                ConnectionFactory:
+                    '!MDBConnectionFactory':
+    '!WebAppContainer':
+appDeployments:
+    Application:
+        '!myear2':
+        yourear:
+            SourcePath: /home/johnny/dimtemp23/sample_app_stage/wlsdeploy/applications/sample_app2.ear
+            ModuleType: ear
+            Target: [ 'cluster-2', 'cluster-1' ]
+        myear:
+            Target: [ 'cluster-2', 'cluster-1' ]
+topology:
+    ServerTemplate:
+        'cluster-1-template':
+            '!ServerStart':
+        'cluster-2-template':
+            Cluster: 'cluster-2'
+            ListenPort: 8001
+            ServerStart:
+                Arguments: [ '-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=6006' ]
+            JTAMigratableTarget:
+                StrictOwnershipCheck: True
+                Cluster: 'cluster-2'
+    Cluster:
+        'cluster-2':
+            DynamicServers:
+                ServerTemplate: 'cluster-2-template'
+                ServerNamePrefix: 'managed-server'
+                DynamicClusterSize: 2
+                MaxDynamicClusterSize: 3
+                CalculatedListenPorts: False
+
+
+1. Model Path: resources-->JMSSystemResource-->MyJmsModule-->JmsResource-->UniformDistributedQueue-->MyUniformDistributedQueue-->ResetDeliveryCountOnForward does not exist in new model but exists in previous model
+
+2. Model Path: appDeployments-->Application-->myear-->ModuleType does not exist in new model but exists in previous model
+
+```
+
+Comparing the new and old models:
+ 
+ 1. Added `cluster-2` and `cluster-2-template` in the topology section
+ 2. Removed `ServerStart` of `cluster-1-template` in the topology section
+ 3. Deployed two applications `yourear` and `myear` in the appDeployments section
+ 4. Removed application `myear` in the appDeployments section
+ 4. Removed `MDBConnectionFactory` from the `MyJmsModule` JMS Module in the resource section
+ 5. Removed `WebAppContainer` in the resource section
+ 6. Removed the attribute `ResetDeliveryCountOnForward` from `MyUniformDistributedQueue`
+ 7. Changed deployment targets for application `myear`
+ 8. Removed the attribute 'ModuleType' for application `myear` 
+ 
+ **Note**:  The `!` is a notation for the deletion of a non-attribute key element from the model. Any missing attribute will 
+ be omitted from the resulting model, but shown as messages in the output.
```

Comment:
Any missing attribute will be omitted from the resulting model, -> but shown as a message in the output. 
(all singular OR all plural)
Missing attributes will be omitted from the resulting model, but shown as messages in the output.